### PR TITLE
Retrieve service by class and properties

### PR DESCRIPTION
**Pull request by stfnsche**
see http://issues.openengsb.org/jira/browse/OPENENGSB-3759

### CODE REVIEW

File: bridge/src/main/java/org/openengsb/loom/java/ProxyConnectorFactory.java [hunk 1]
```diff
@@ -40,16 +41,28 @@ public ProxyConnectorFactory(ProtocolHandler remoteConfig, String principal, Cre
         this.credentials = credentials;
     }
 
+    public <T> T getRemoteProxy(Class<T> serviceClass) {
```
Comment (kristofmeixner) [inline bridge/src/main/java/org/openengsb/loom/java/ProxyConnectorFactory.java:44]: Thing that bothers me a bit is if someone passes `null`  here.

File: bridge/src/main/java/org/openengsb/loom/java/jms/JmsProtocolHandler.java [hunk 2]
```diff
@@ -119,7 +118,7 @@ public void onMessage(Message message) {
     private QueueMap<String, Message> replyMessageQueue = new QueueMap<String, Message>();
 
     public JmsProtocolHandler(String baseURL, String applicationid) throws JMSException {
-        this.applicationid = applicationid;
```
Comment (kristofmeixner) [inline bridge/src/main/java/org/openengsb/loom/java/jms/JmsProtocolHandler.java:122]: If you changed to camelcase why didn't you change the passed parameter?

**Comment by kristofmeixner** [comment]
Why are the issue numbers missing at your commit messages?

File: bridge/src/main/java/org/openengsb/loom/java/ProxyConnectorFactory.java (hunk 1 above)
Comment (stfnsche) [inline bridge/src/main/java/org/openengsb/loom/java/ProxyConnectorFactory.java:44]: I could check against null and explicitly throw a NPE, or do nothing and a NPE will be implicitly thrown 2 lines below (where the classloader is fetched). 
Or what solution do you have in mind?

**Comment by stfnsche** [comment]
Fixed reported review issues. Please, take another look and merge if everything okay.

File: bridge/src/main/java/org/openengsb/loom/java/ProxyConnectorFactory.java (hunk 1 above)
Comment (stfnsche) [inline bridge/src/main/java/org/openengsb/loom/java/ProxyConnectorFactory.java:44]: Added a null check in ServiceIndentifier.

**Pull request merged**